arkouda/categorical.py

```diff
@@ -843,13 +843,30 @@ def _get_grouping_keys(self):
         """
         return [self.codes]
 
-    def argsort(self):
-        # __doc__ = argsort.__doc__
+    def argsort(self, ascending=True):
+        """
+        Return the permutation that sorts the Categorical.
+
+        Parameters
+        ----------
+        ascending: bool = True
+
+        Returns
+        -------
+        pdarray, int64
+            The indices that sort the Categorical.
+        """
+        from arkouda import arange
+
         idxperm = argsort(self.categories)
         inverse = zeros_like(idxperm)
         inverse[idxperm] = arange(idxperm.size)
         newvals = inverse[self.codes]
-        return argsort(newvals)
+        sorted_array = argsort(newvals)
+        if ascending is True:
+            return sorted_array
+        else:
+            return sorted_array[arange(sorted_array.size - 1, -1, -1)]
```

Comment:
FWIW, that implementation of flip is not very efficient because it has to generalize to any number of dimensions. I'd expect it would give you about the same performance as `sorted_array[::-1]`.

A good strategy might be to use `flipMsg` for now, where I'll probably add a more efficient 1D path at some point in the future. And then we won't have to do anything for `sort_values` to take advantage of those performance improvements when that happens. (This will require uncommenting [this line](https://github.com/Bears-R-Us/arkouda/blob/5b3ee154ff93858f96899b13b9cfc121088258d8/ServerModules.cfg#L22) to always build ManipulationMsg — which is a switch we were going to have to flip at some point to support other numpy-like functionality in the main arkouda api).

Reply:
makes sense! I agree,I think enabling `ManipulationMsg` in the server modules config file and using flip here is the way to go